query-engine/connectors/sql-query-connector/src/filter/visitor.rs

```diff
@@ -1266,7 +1276,10 @@ impl JsonFilterExt for (Expression<'static>, Expression<'static>) {
         match (value, target_type) {
             // string_contains (value)
             (ConditionValue::Value(value), JsonTargetType::String) => {
-                let contains = expr_string.like(format!("%{value}%"));
+                let contains = match query_mode {
+                    QueryMode::Default => expr_string.like(format!("%{value}%")),
+                    QueryMode::Insensitive => expr_string.compare_raw("ILIKE", format!("%{value}%")),
```

Comment:
The `ILIKE` comparison operator is only available on PostgreSQL and CockroachDB.
Since Prisma supports several other database providers (like SQLite, MySQL, MongoDB), we should steer away from using `raw` operators directly.

Usually, we rely on connector-specific primitives wrapped in a common abstraction (`quaint::connector::queryable::Queryable`).

But in this case, the way forward is probably to change the implementations of the `quaint::visitor::Visitor` trait to align `visit_json_extract` and similar methods to https://github.com/prisma/prisma/issues/7390#issuecomment-2244497806, lowering the case of both compared values when `mode: 'insensitive'` is applied.

Reply:
Nice catch, thank you!